src/Yoti/ActivityDetails.php

```diff
@@ -103,29 +106,17 @@ public static function constructFromAttributeList(AttributeList $attributeList,
                 $attrs[$attrName] = $item->getValue();
             }
             // Build attribute object for user profile
-            $attrValue = $item->getValue();
-            // Convert structured_postal_address value to an Array
-            if ($attrName === 'structured_postal_address') {
-                $attrValue = json_decode($attrValue, TRUE);
-            }
-
             $attributeAnchors = $anchorProcessor->process($item->getAnchors());
-            $attribute = new Attribute(
-                $attrName,
-                $attrValue,
-                $attributeAnchors['sources'],
-                $attributeAnchors['verifiers']
-            );
-
-            $profileAttributes[$attrName] = $attribute;
+            $yotiAttribute = self::getYotiAttribute($item, $attributeAnchors, $attrName);
+            $profileAttributes[$attrName] = $yotiAttribute;
```

Comment:
Should all this be inside a null check on _$yotiAttribute_?  I don't see any point adding a _null_ into the _$profileAttributes_ map.

Reply:
I thought that could be confusing to users.
In this case the Attribute `document_details' is set in Dashboard and returned by connect, but because the value is invalid we are returning NULL. 
If we don't add it to the `$profileAttributes` map, the  user will think this attribute is not being sent by connect.
I am happy to not include it , if we think that's the way to go.